include/cand/can.hpp

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <iomanip> 
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+#include <unistd.h>
+
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
+#include <linux/can.h>
+#include <linux/can/raw.h>
+
+#ifndef LINUX_CAN_H
+#define LINUX_CAN_H
+
+class CAN{
+    private:
+
+        int file_descriptor;
+        struct can_frame frame;
+        std::string interface_name;
+
+        friend CAN& operator<<(CAN&, struct can_frame *);
+
+
+    public:
+        CAN(std::string name);
+
+        int socket_write(struct can_frame *);
+
+        int socket_close();
+
+};
+std::ostream& operator<<(std::ostream&,struct can_frame&);
```

Comment:
Isso realmente pertence fora da classe?

Reply:
Olá, Maíra. Creio que pertença fora da classe sim, visto que é apenas um print de um frame por fora. A não ser que a intenção seja ler o frame que está dentro do CAN, se for isso peço até perdão, vi agora que o frame dentro da classe nunca é usado. O frame é colocado dentro do CAN no write ou no read?

Reply:
Basta você colocar `friend std::ostream& operator<<(std::ostream&,struct can_frame&);` nos métodos públicos da sua classe